docs/pages/guides/server-components.mdx

```diff
@@ -0,0 +1,400 @@
+    },
+  });
+
+  // ...
+  return <div />;
+}
+```
+
+You can define the secret in your `.env` file:
+
+```env
+SECRET=123
+```
+
+## Platform detection
+
+To detect which platform your code is bundled for, use the `process.env.EXPO_OS` envrionment variable. For example `process.env.EXPO_OS === 'ios'`. Prefer this to `Platform.OS` as `react-native` is not fully optimized for React Server Components yet and therefore won't work as expected.
+
+You can detect if code is running in the server by performing a `typeof window === 'undefined'` check. This will always return `true` on client devices and `false` on the server.
+
+## Testing with jest
+
+Library authors can test their modules support server components by using `jest-expo`. Learn more in the [testing react server components guide](/guides/testing-rsc).
+
+## Known limitations
+
+> This is a very early technical preview that we're actively developing.
+
+- There is currently no stack routing. The custom layouts `Stack`, `Tabs`, `Drawer` do not support server components yet.
+- Expo Go does not support the new architecture yet, meaning you cannot use server components in Expo Go.
+- Expo Snack does not support bundling server components.
+- EAS Update does not work with server components yet.
+- DOM components cannot use server components yet.
+- Production deployment is very limited and not recommended yet.
+- Server Rendering RSC payloads to HTML is not supported yet.
+- `generateStaticParams` is not yet supported.
+- HTML `form` integration with Server Actions is not supported yet (this partially works automatically, but data is not encrypted).
+
+## Deployment
+
+> Avoid using this feature in production as it's not stable yet. We'll streamline all of this to a single command in the future.
+
+### Web
+
+To deploy your website to production, run: `npx expo export -p web`.
+
+You can host the website locally with the [express server adapter](https://docs.expo.dev/router/reference/api-routes/#express).
+
+### Native
+
+Unlike standard native builds, React Server Components requires an additional server to be generated and hosted. This server is responsible for rendering the server components and serving the data to the native app. We'll generate this server code during the native build in order to optimize and link static assets to the binary.
+
+First, set the `origin` in your `app.json` to the URL of your server:
+
+```json app.json
+{
+  "expo": {
+    "plugins": [
+      [
+        "expo-router",
+        {
+          // Setting this to a local server URL.
+          "origin": "http://localhost:3000"
+        }
+      ]
+    ]
+  }
+}
+```
+
+To deploy your native app to production, you'll need to build it for production with `npx expo run:ios --configuration Release` or `npx expo run:android --variant release`. This will generate the required server code to `./.expo/server/<platform>` along with building and embedding any assets which could potentially be local-first into the binary.
```

Comment:
```suggestion
To deploy your native app to production, you'll need to build it for production with `npx expo run:android --variant release` or `npx expo run:ios --configuration Release`. This will generate the required server code to `./.expo/server/<platform>` along with building and embedding any assets that could potentially be local-first into the binary.
```